docs/website/src/md-image/md-image.ts

```diff
@@ -0,0 +1,118 @@
+import { attr, boolAttr, ESLBaseElement, listen } from '@exadel/esl/modules/esl-base-element/core';
+import { DeviceDetector, getOffsetPoint, getTouchPoint, Point } from '@exadel/esl';
+import { memoize } from '@exadel/esl/modules/esl-utils/decorators/memoize';
+
+export class MdImage extends ESLBaseElement {
+  static is = 'md-image';
+  static zoomClass: string = 'md-image-zoom';
+
+  @attr({ dataAttr: true })
+  public src: string;
+
+  @boolAttr()
+  public inZoom: boolean;
+  @boolAttr()
+  public inDrag: boolean;
+
+  public scale: number = 2;
+  protected startPoint: Point;
+  protected startPosition: Point;
+
+  @memoize()
+  get $image() {
+    const originalImage = document.createElement('esl-image');
+    originalImage.setAttribute('data-src', this.src);
+    originalImage.setAttribute('lazy', '');
+    originalImage.setAttribute('mode', 'fit');
+    return originalImage;
+  }
+
+  @memoize()
+  get $closeButton() {
+    const button = document.createElement('button');
+    button.onclick = this.onClose.bind(this);
+    button.setAttribute('class', 'close-button');
+    button.innerText = 'X';
+    button.style.display = 'none';
+    return button;
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.appendChild(this.$image);
+    this.appendChild(this.$closeButton);
+  }
+
+  @listen('click')
+  protected onClick(e: MouseEvent): void {
+    if (this.inDrag || this.inZoom) return;
+    this.inZoom = true;
+
+    this.$image.style.transform = `scale(${this.scale})`;
+    this.style.width = `${this.$image.offsetWidth}px`;
+    this.style.height = `${this.$image.offsetHeight}px`;
+
+    this.$closeButton.style.display = 'block';
+    this.$$on('pointerdown', this.onPointerDown);
+  }
+
+  protected onClose(e: MouseEvent): void {
+    e.stopPropagation();
+    if (!this.inZoom) return;
+    this.inZoom = false;
+
+    this.$image.style.removeProperty('transform');
+    this.style.width = 'auto';
+    this.style.height = 'auto';
+
+    this.$closeButton.style.display = 'none';
+
+    this.$$off(this.onPointerDown);
```

Comment:
I'd recommend to create a method `reset` / `cancelZoom` or something like that to separate everything that relates to event handling from exact action